### PR TITLE
Adding Multithread test

### PR DESCRIPTION
**Pull request by crystalhsieh**
Testing threading using Python rather than the original bash script by running the /hold handler (which stays alive forever) and then trying to call a /static handler in a separate thread.

### CODE REVIEW

File: multithread_test.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+from os import system
+from threading import Thread
+import socket
+import sys
+import subprocess
+import time
+
+PORT = 2020
+BUFFER_LEN = 1024
+HOLD_REQUEST = "GET /hold HTTP/1.1"
```
Comment (jasonsliu) [inline multithread_test.py:12]: Technically supposed to have CRLF but also it doesn't actually matter according to the holding handler code huh.

File: multithread_test.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+STATIC_REQUEST = "GET /static/test.html HTTP/1.1\r\n\r\n"
+EXPECTED_STATIC_RESPONSE = "HTTP/1.1 200 OK\r\nContent-Type: text/html\r\n\r\n<!DOCTYPE html>\r\n<html>\r\n<head>\r\n<title>Server Test</title>\r\n</head>\r\n<body>\r\n<h1>Testing testing 1 2 3...</h1>\r\n<p>Looks like the server is working!</p>\r\n</body>\r\n</html>"
```
Comment (jasonsliu) [inline multithread_test.py:14]: Might be safer if we directly read the file, but this is okay for now

File: multithread_test.py (hunk 1 above)
Comment (jasonsliu) [reply multithread_test.py:12]: Oh I just noticed you send it separately.. on purpose

File: multithread_test.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+	two_connections_results = two_connections_test()
+	time.sleep(2)
+
+	if (two_connections_results == 0):
+		print "THREAD SUCCESS"
+		exit(0)
+	else:
+		print "THREAD FAILURE"
+		exit(1)
```
Comment (jasonsliu) [inline multithread_test.py]: Please add newline

**Review by jasonsliu — APPROVED** [review]
If you have time, add exit 0 and exit 1 to the shell scrip integration test !

File: multithread_test.py (hunk 2 above)
Comment (crystalhsieh) [reply multithread_test.py:14]: the part that i was worried about was that I would still need the header of "HTTP/1.1 200 OK" from the expected response.

File: multithread_test.py (hunk 3 above)
Comment (crystalhsieh) [reply multithread_test.py]: done!

**Pull request merged**